apps/dgst.c

```diff
@@ -435,21 +421,20 @@ int dgst_main(int argc, char **argv)
         }
         ret = 0;
         for (i = 0; i < argc; i++) {
-            int r;
             if (BIO_read_filename(in, argv[i]) <= 0) {
                 perror(argv[i]);
                 ret++;
                 continue;
             } else {
-                r = do_fp(out, buf, inp, separator, out_bin, xoflen,
-                          sigkey, sigbuf, siglen, sig_name, md_name, argv[i]);
+                ret = do_fp(out, buf, inp, separator, out_bin, xoflen,
```

Comment:
Isn't this changing semantics? This will happily overwrite the ret if one do_fp() call passes where previously there might have been failures already.

Reply:
Oh, good spotting!
Fixed.